examples/shapley_value_with_customized_featurizers.ipynb

```diff
@@ -0,0 +1,1176 @@
+    "    test_loader = data.build_dataloader(test_dset, shuffle=False)\n",
+    "\n",
+    "    with torch.inference_mode():\n",
+    "        trainer = pl.Trainer(\n",
+    "            logger=None,\n",
+    "            enable_progress_bar=False,\n",
+    "            accelerator=\"cpu\",\n",
+    "            devices=1\n",
+    "        )\n",
+    "        test_preds = trainer.predict(mpnn, test_loader)\n",
+    "    return test_preds[0][0]"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 210,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Prediction with all atoms and bonds kept: tensor([2.2535])\n",
+      "Prediction with all atoms and bonds removed: tensor([2.1880])\n"
+     ]
+    },
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "/home/oscarwu/code/chemprop_developing/chemprop/data/dataloader.py:52: UserWarning: Dropping last batch of size 1 to avoid issues with batch normalization (dataset size = 1, batch_size = 64)\n",
+      "  warnings.warn(\n",
+      "/home/oscarwu/mambaforge/envs/chemprop_v2/lib/python3.12/site-packages/lightning/fabric/plugins/environments/slurm.py:204: The `srun` command is available on your system but is not used. HINT: If your intention is to run Lightning on SLURM, prepend your python command with `srun` like so: srun python /home/oscarwu/mambaforge/envs/chemprop_v2/lib/python ...\n",
+      "/home/oscarwu/mambaforge/envs/chemprop_v2/lib/python3.12/site-packages/lightning/pytorch/trainer/connectors/data_connector.py:424: The 'predict_dataloader' does not have many workers which may be a bottleneck. Consider increasing the value of the `num_workers` argument` to `num_workers=63` in the `DataLoader` to improve performance.\n"
+     ]
+    }
+   ],
+   "source": [
+    "# example prediction with different keep_atoms and keep_bonds\n",
+    "\n",
+    "# keep all atoms and bonds\n",
+    "keep_atoms_0 = [True] * n_atoms\n",
+    "keep_bonds_0 = [True] * n_bonds\n",
+    "\n",
+    "# remove all atoms and bonds\n",
+    "keep_atoms_1 = [False] * n_atoms\n",
+    "keep_bonds_1 = [False] * n_bonds\n",
+    "\n",
+    "pred_0 = get_predictions(keep_atoms_0, keep_bonds_0, test_mol)\n",
+    "pred_1 = get_predictions(keep_atoms_1, keep_bonds_1, test_mol)\n",
+    "\n",
+    "print(f\"Prediction with all atoms and bonds kept: {pred_0}\") # expected 2.2535\n",
+    "print(f\"Prediction with all atoms and bonds removed: {pred_1}\") # expected 2.1880"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 211,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# An example wrapper class for use as the model input in SHAP explainer\n",
+    "class MoleculeModelWrapper:\n",
+    "    def __init__(self, mol: str, n_atoms: int, n_bonds: int):\n",
+    "        self.mol = mol\n",
+    "        self.n_atoms = n_atoms\n",
+    "        self.n_bonds = n_bonds\n",
```

Comment:
If you passed in `mol` as an `rdkit.Chem.mol` object then you could get `n_atoms` and `n_bonds` inside here automatically instead of needing the user to make the `mol` object, calculate `n_atoms` and `n_bonds` and then pass them in.